### DISCUSSION TITLE
Is there a sample code for CoAP secure? (Use tinydtls)

### DISCUSSION THREAD

**Discussion by leolin0803** [contiki-ng/contiki-ng#2931] [General]
Hi I am a beginner in contiki, and I want to use two terminals for CoAPs (CoAP secure) data transmission, but I don’t know how to do it.
I have searched some contiki github encrypted with tinydtls, but running their sample code can not connect the server and client.
I don’t know if there is a sample program for reference.

Another problem is that when running your coap example code **coap-example-server.native**, a virtual network interface tun0 will be opened, and i can connect normally when using libcoap. However, the error message "Failed to open tun device" will appear when connecting with the sample code **coap-example-client.native.**
Is there any error in my settings?
terminal 1
run commend --> sudo ./coap-example-server.native and input my password of linux.

terminal 2
run commend --> sudo ./coap-example-client.native and input my password of linux.

Thanks you in advance.

**Comment by damien-v** [comment]
Hello, what are you using as a machine to run the example and could you share some output? Have you tried using the contiker/contiki-ng docker image?
And just to be sure, have you changed the server ip address in the client example?
```c
/* FIXME: This server address is hard-coded for Cooja and link-local for unconnected border router. */
#define SERVER_EP "coap://[fe80::212:7402:0002:0202]"
```
